### ISSUE TITLE
Lag - Delay

### ISSUE THREAD

**Issue by LiasOne** [Tudmotu/gnome-shell-extension-clipboard-indicator#461]
After fixing the #445

When using the shortcut to change to the next entry the Ctrl+V is pasting the previous entry.  It needs 2 sec to paste the selected entry. Lag - delay 2 sec.

Arch Linux
Gnome 46
Wayland

**Comment by Tudmotu** [comment]
Do you have the "move item to top" setting enabled?
![image](https://github.com/Tudmotu/gnome-shell-extension-clipboard-indicator/assets/506487/b7fe8108-a52d-4bb8-885b-d68031c8d73a)

**Comment by LiasOne** [comment]
Yes, I have this enabled. I have had it before but this is the first time that this lag has been happening.

**Comment by Tudmotu** [comment]
The lag is there because otherwise, when using the next/previous shortcuts, you get into a loop between the first two items:
1. You select something in the clipboard, it moves to the first position
2. Now you use the shortcut to select the next entry
3. The 2nd entry immediately moves to the top of the list
4. Now the item you first selected, is the 2nd item
5. You use the short to select the next item again
6. You are stuck in a loop

The delay allows you to rapidly move between the items before they are moved to the top.

There could maybe be a better way of achieving this.

Now, I am not 100% sure that your issue is not something else.

Can you please disable this option and see if the delay persists?

**Comment by LiasOne** [comment]
As I said in the previous comment I have disabled the option to move item to the top after your recommendation.  And is working smoothly.

Thank you very much for your support and for this awesome extension. It helps me a lot in my daily job.

**Comment by Tudmotu** [comment]
> As I said in the previous comment I have disabled the option to move item to the top after your recommendation. And is working smoothly.
> 
> Thank you very much for your support and for this awesome extension. It helps me a lot in my daily job.

Ah, thank you! Sorry, I missed the last part 🙂 
Thank you for confirming 🙏 

I am gonna leave this issue open, in case anyone comes up with an idea to prevent the looping issue from happening without introducing a timeout between selections.
